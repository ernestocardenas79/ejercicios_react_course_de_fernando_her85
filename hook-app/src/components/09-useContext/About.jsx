import { useContext } from 'react';
import { UserContext } from './UserContext';

export const About = () => {
  const {user, setUser} = useContext(UserContext);

  return (
    <div>
      <h1>About</h1>
      <hr/>

      <pre>
        {JSON.stringify(user,null, 4)}
      </pre>

      <button className="btn btn-warning" onClick={()=>setUser({})}>Logout</button>
    </div>
  );
};
