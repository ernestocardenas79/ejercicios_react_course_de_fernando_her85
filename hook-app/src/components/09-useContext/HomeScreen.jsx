import { UserContext } from './UserContext';
import { useContext } from 'react';

export const HomeScreen = () => {
  const {user} = useContext(UserContext);

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre aria-label='pre'>
        { JSON.stringify(user, null, 4)}
      </pre>
    </div>
  );
};
