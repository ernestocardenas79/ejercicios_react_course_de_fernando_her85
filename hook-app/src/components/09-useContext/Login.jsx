import { useContext } from 'react';
import { UserContext } from './UserContext';

export const Login = () => {
  const {setUser} = useContext(UserContext);
  const user = {
    id:774,
    name:'Ernesto'
  }
  return (
    <div>
      <h1>Login</h1>
      <button className="btn btn-primary" onClick={()=>setUser(user)}>
        Login
      </button>
    </div>
  );
};
