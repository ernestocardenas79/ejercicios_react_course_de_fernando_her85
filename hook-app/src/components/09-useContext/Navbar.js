import React from "react";
import { NavLink } from "react-router-dom";

export const Navbar = () => {
  return (
    <nav class="navbar navbar-expand-sm navbar-dark bg-dark">
      <h1>useContext</h1>

      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="./"
            >
              Home <span class="sr-only">(current)</span>
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="./about"
            >
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="./login"
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
