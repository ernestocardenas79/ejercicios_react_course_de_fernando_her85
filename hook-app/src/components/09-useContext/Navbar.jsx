import { NavLink, Link } from 'react-router-dom';

export const Navbar = () => {
  return (
    <nav className='navbar navbar-expand-sm navbar-dark bg-dark rounded-3'>
      <Link className="navbar-brand" to="/">useContext</Link>

      <div className='collapse navbar-collapse' id='navbarNav'>
        <ul className='navbar-nav'>
          <li className='nav-item'>
            <NavLink
              className={({isActive})=>`nav-link ${ isActive? 'active':''}`}
              to='/'
            >
              Home <span className='sr-only'>(current)</span>
            </NavLink>
          </li>
          <li className='nav-item'>
            <NavLink
               className={({isActive})=>`nav-link ${ isActive? 'active':''}`}
              to='about'
            >
              About
            </NavLink>
          </li>
          <li className='nav-item'>
            <NavLink
              className={({isActive})=>`nav-link ${ isActive? 'active':''}`}
              to='login'
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
