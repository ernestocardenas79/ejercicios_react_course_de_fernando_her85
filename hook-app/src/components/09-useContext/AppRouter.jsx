import { About } from './About';
import { Login } from './Login';
import { HomeScreen } from './HomeScreen';
import {
  Route,
  Navigate,
  Routes,
} from 'react-router-dom';

export const AppRouter = () => {
  return (
    <Routes>
      <Route  path='/' element={<HomeScreen/>} />
      <Route  path='about' element={<About/>} />
      <Route  path='login' element={<Login/>} />
      <Route  path='/*' element={<Navigate to="/login"/>} />
    </Routes>
  );
};
