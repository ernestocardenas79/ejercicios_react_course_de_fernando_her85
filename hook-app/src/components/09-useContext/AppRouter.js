import React from 'react';
import { About } from './About';
import { Login } from './Login';
import { HomeScreen } from './HomeScreen';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import { Navbar } from './Navbar';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <div className="container">
        <Switch>
          <Route exact path='/about' component={About} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/' component={HomeScreen} />
          <Redirect component={HomeScreen} />
        </Switch>
        </div>
      </div>
    </Router>
  );
};
