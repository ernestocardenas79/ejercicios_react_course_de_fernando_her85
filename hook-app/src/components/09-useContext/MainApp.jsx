import { Navigate, Route, Routes } from 'react-router-dom';
import { UserContext } from './UserContext';
import { useState } from 'react';
import { HomeScreen } from './HomeScreen';
import { About } from './About';
import { Login } from './Login';
import { Navbar } from './Navbar';

import './styles.css'

export const MainApp = () => {
 const [user, setUser] = useState({});

  return (
    <UserContext.Provider value={{user, setUser}}>
      <Navbar />
      <hr />
              
      <Routes>
          <Route  path='/' element={<HomeScreen/>} />
          <Route  path='about' element={<About/>} />
          <Route  path='login' element={<Login/>} />
          <Route  path='/*' element={<Navigate to="/login"/>} />
      </Routes>   
    </UserContext.Provider>
  );
};


   