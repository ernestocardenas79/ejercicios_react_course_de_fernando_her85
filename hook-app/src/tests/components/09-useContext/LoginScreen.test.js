import React from 'react';
import { Login } from '../../../components/09-useContext/Login';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Preubas en <LoginScreen />', () => {
  const setUser = jest.fn(() => {});
  const user = {
    name: 'Ernesto',
    id: 774,
  };

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <Login />
    </UserContext.Provider>
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de llamarse la funcion setUser con lo argumentos esperados', () => {
    wrapper.find('button').prop('onClick')();

    expect(setUser).toHaveBeenCalledWith(user);
  });
});
