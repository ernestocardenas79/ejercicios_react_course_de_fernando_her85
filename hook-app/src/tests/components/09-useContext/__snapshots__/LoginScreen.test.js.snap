// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preubas en <LoginScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Login
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
